kasa/cli.py

```diff
@@ -90,19 +90,62 @@ def wrapper(message=None, *args, **kwargs):
 pass_dev = click.make_pass_decorator(Device)
 
 
-class ExceptionHandlerGroup(click.Group):
-    """Group to capture all exceptions and print them nicely.
+def CatchAllExceptions(cls):
+    """Capture all exceptions and prints them nicely.
 
-    Idea from https://stackoverflow.com/a/44347763
+    Idea from https://stackoverflow.com/a/44347763 and
+    https://stackoverflow.com/questions/52213375
     """
 
-    def __call__(self, *args, **kwargs):
-        """Run the coroutine in the event loop and print any exceptions."""
-        try:
-            asyncio.get_event_loop().run_until_complete(self.main(*args, **kwargs))
-        except Exception as ex:
-            echo(f"Got error: {ex!r}")
+    def _handle_exception(cmd, info_name, exc):
+        if isinstance(exc, click.ClickException):
+            raise
+        echo(f"Kasa:: Command line: {info_name} {cmd._masked_args}")
+        echo(f"Kasa:: Raised error: {exc}")
+        if cmd._debug:
             raise
+        echo("Run with --debug enabled to see stacktrace")
+        sys.exit(1)
+
+    class Cls(cls):
+        _debug = False
+        _masked_args = None
+
+        def _parse_args(self, args):
+            masked_args = []
+            debug = False
+            pw_index = un_index = None
+            for index, arg in enumerate(args):
+                if arg in ["--password", "-p"]:
+                    pw_index = index + 1
+                if arg in ["--username", "-u"]:
+                    un_index = index + 1
```

Comment:
Hmh, what's the reason for this? Only to display it in the cli output? I'm not sure if we want such behavior in the first place, I'd rather keep it as simple as possible.

Reply:
Removed